fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -325,26 +328,26 @@ private CompletableFuture<Void> handleCrossPartitionDiscontinuity() {
         // with respect to the continuation, do that now:
         return partitioner.getPartitionMetaInfoById(Objects.requireNonNull(partitionId), groupingKey).thenCompose(partitionInfo -> {
             // the boundaries of the partition may have changed (due to re-balancing, explicit doc removal etc.)
-            long currentFrom = LucenePartitioner.getFrom(partitionInfo);
-            long currentTo = LucenePartitioner.getTo(partitionInfo);
-            partitionTimestamp = currentFrom;
+            Tuple currentFrom = Tuple.fromBytes(partitionInfo.getFrom().toByteArray());
+            Tuple currentTo = Tuple.fromBytes(partitionInfo.getTo().toByteArray());
+            partitionKey = currentFrom;
 
             final CompletableFuture<Void> getProperContinuationPartition;
 
-            // next, if the current partition's new boundary excludes the continuation's "search after" timestamp,
-            // we need to change the current partition to the one that does contain this timestamp.
-            if (searchAfterTimestamp != null && (searchAfterTimestamp < currentFrom || searchAfterTimestamp > currentTo)) {
+            // next, if the current partition's new boundary excludes the continuation's "search after" partitioning key,
+            // we need to change the current partition to the one that does contain this partitioning key.
+            if (searchAfterPartitioningKey != null && (searchAfterPartitioningKey.compareTo(currentFrom) < 0 || searchAfterPartitioningKey.compareTo(currentTo) > 0)) {
```

Comment:
Can/should this use `LucenePartitioner.isOlderThan` or `isNewerThan`?

Reply:
yes... I wrote this before adding those functions. Will update.